app/controllers/comments_controller.rb

```diff
@@ -1,2 +1,28 @@
 class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @user = User.find(1)
+    @new_comment = @user.comments.new(
+      text: comment_params,
+      author_id: @user.id,
+      post_id: @post.id
+    )
+    @new_comment.post_id = @post.id
+    @new_comment.update_comments_counter
+    if @new_comment.save
+      redirect_to "/users/#{@post.author_id}/posts/#{@post.id}"
+    else
+      render text: 'Could not save post'
+    end
+  end
+
```

Comment:
- [ ] kindly fix this issue that is rendering the complete hash in the view. you can fix it by extracting the text in the params when creating. Hint :point_down: 

```rb
    @new_comment = @user.comments.new(
      text: comment_params[:text],
      author_id: @user.id,
      post_id: @post.id
    )
```


![image](https://user-images.githubusercontent.com/65068771/161357614-049b2f6c-2cef-4340-b75c-ccf1b53b2360.png)